libs/rest/src/struct/Bucket.ts

```diff
@@ -77,47 +75,29 @@ export class Bucket {
     if (limit) state.limit = Number(limit);
     if (remaining) state.remaining = Number(remaining);
     if (resetAfter) state.timeout = Number(resetAfter) * 1000;
+    this.rest.emit('response', req, res.clone(), state);
 
     await this.mutex.set(this.route, state);
 
     if (res.status === 429) {
-      const retry = res.headers.get('Retry-After');
+      const retry = res.headers.get('retry-after');
       /* istanbul ignore next */
-      const retryAfter = retry ? Number(retry) * 1000 : 0;
+      const retryAfter = Number(retry ?? 1) * 1000;
 
-      this.manager.emit('ratelimit', this.route, req.path, false, retryAfter);
+      this.rest.emit('ratelimit', this.route, req.path, false, retryAfter);
 
       await this.mutex.set(this.route, { timeout: retryAfter });
-      return this._retry<T, D, Q>(req);
+      return Promise.reject(new CordisRestError('rateLimited'));
     } else if (res.status >= 500 && res.status < 600) {
-      await halt(1000);
-      return this._retry<T, D, Q>(req);
+      return Promise.reject(new CordisRestError('internal'));
     } else if (!res.ok) {
-      throw new HTTPError(res.clone(), await res.text());
+      return Promise.reject(new HTTPError(res.clone(), await res.text()));
     }
 
-    let final: any;
-
-    if (res.headers.get('content-type') === 'application/json') final = await res.json();
-    else final = await res.blob();
-
-    this.manager.emit('response', req, final, state);
-    return final;
-  }
-
-  /**
-   * Retries to make a request after failure
-   * @param req Request options
-   * @param res Response given
-   */
-  private _retry<T, D, Q>(req: DiscordFetchOptions<D, Q>): Promise<T> {
-    if (req.failures) req.failures++;
-    else req.failures = 1;
-
-    if (req.failures > this.manager.retries) {
-      throw new CordisRestError('retryLimitExceeded', `${req.method.toUpperCase()} ${req.path}`, this.manager.retries);
+    if (res.headers.get('content-type')?.startsWith('application/json')) {
+      return res.json();
     }
 
-    return this.make(req);
+    return res.blob() as any;
```

Comment:
```suggestion
    return res.blob() as T;
```

Method signature is `Promise<T>` so it should be returned accordingly.

Reply:
`as T` won't do. That `blob()` call should *never* happen. It's only there as more of a safety meassure in case the library ever runs out of date with Discord's API and they have a route that doesn't output JSON for whatever reason - so typings just can't be accurate here which is something I'm willing to sacrifice.

Could cast `as unknown as T` though since `T` isn't assignable to `Blob`.